lib/active_merchant/billing/gateways/cecabank/cecabank_json.rb

```diff
@@ -206,7 +206,18 @@ def add_stored_credentials(post, creditcard, options)
         params[:inicioRec] = initiator
         if initiator == :S
           requires!(options, :recurring_frequency)
-          params[:finRec] = options[:recurring_end_date] || strftime_yyyymm(creditcard)
+          fin_rec = options[:recurring_end_date]
+          params[:finRec] =
+            if fin_rec&.match?(/^\d{8}$/)
+              fin_rec
+            elsif fin_rec&.match?(/^\d{6}$/)
+              year = fin_rec[0, 4].to_i
+              month = fin_rec[4, 2].to_i
+              last_day = Date.civil(year, month, -1).day
+              sprintf('%04d%02d%02d', year, month, last_day)
+            else
+              strftime_yyyymmdd_last_day(creditcard)
+            end
```

Comment:
@aaqibrashidmir I don't think we need to format the date passed in the options.
```suggestion
          fin_rec = options[:recurring_end_date] || strftime_yyyymmdd_last_day(creditcard)
```